packages/components/src/presentation-components/tree/controlled/UsePresentationTreeState.ts

```diff
@@ -221,7 +233,7 @@ function useTreeState(props: UseTreeStateProps) {
   useEffect(() => {
     const { treeModel, ...providerProps } = props.treeStateProps;
     const modelSource = new TreeModelSource(new MutableTreeModel(treeModel));
-    const dataProvider = new PresentationTreeDataProvider(providerProps);
+    const dataProvider = new PresentationTreeDataProvider({ ...providerProps, onHierarchyLimitExceeded: onHierarchyLimitExceededRef.current });
     const pagedLoader = new PagedTreeNodeLoader(dataProvider, modelSource, providerProps.pagingSize);
     const nodeLoader = new ReportingTreeNodeLoader(pagedLoader, () => onNodeLoadedRef.current);
```

Comment:
I missed this during PR review but looking at this now I think it would be more readable if this was not a getter for `onNodeLoaded` callback. I would suggest changing it into `(nodeLoadedProps) => onNodeLoadedRef.current?.(nodeLoadedProps)`